tests/test_dump_file.py

```diff
@@ -187,3 +187,71 @@ def test_copy_from(snap, tmp_path):
     assert numpy.all(read_snap.molecule == [0, 1, 2])
     assert read_snap.has_charge()
     assert numpy.allclose(read_snap.charge, [0, 1, -1])
+
+
+def test_faulty_dump_schema(snap, tmp_path):
```

Comment:
To further reduce code duplication, can you also parametrize this on the different parts of the schema you're trying to check for? For example:
```suggestion
@pytest.mark.parametrize(
    "schema",
    ["x", "x y", "vx", "vx vy", "ix", "ix iy"],
    ids=["position-1d",  "position-2d", "velocity-1d", "velocity-2d", "image-1d", "image-2d"])
def test_faulty_dump_schema(snap, tmp_path, schema):
```
This would also let you test for 1d values and not just 2d values tripping the error!

Two small changes are needed below as well. I will mark them there.